code/modules/surgery/organs/internal/stomach/_stomach.dm

```diff
@@ -199,6 +208,133 @@
+			var/obj/item/as_item = nugget
+			damage = as_item.w_class * 2
+		else if (isliving(nugget))
+			var/mob/living/as_living = nugget
+			damage = as_living.mob_size * 5
+		owner.apply_damage(damage, BRUTE, BODY_ZONE_CHEST, wound_bonus = CANT_WOUND, wound_clothing = FALSE)
+	return emptied
+
+/obj/item/organ/stomach/on_life(seconds_per_tick, times_fired)
+	. = ..()
+	if (!owner || SSmobs.times_fired % 3 != 0)
+		return
+
+	if (!length(stomach_contents))
+		return
+
+	var/obj/item/bodypart/chest/chest = owner.get_bodypart(BODY_ZONE_CHEST)
+	var/datum/wound/slash/flesh/slash = chest.get_wound_type(/datum/wound/slash/flesh)
+	// A chance to spill out all the contents
+	if (cut_open_damage && slash?.severity >= WOUND_SEVERITY_CRITICAL)
+		if (SPT_PROB(chest.get_damage(), seconds_per_tick * 3))
+			var/emptied = empty_contents()
+			if (emptied > 0)
+				owner.apply_damage(emptied * 5, BRUTE, BODY_ZONE_CHEST, wound_bonus = CANT_WOUND, wound_clothing = FALSE)
+				playsound(get_turf(src), 'sound/effects/splat.ogg', 50)
+				owner.visible_message(span_danger("Contents of [owner]'s intestines spill out from a huge cut in [owner.p_their()] [chest]!"),
+					span_userdanger("Contents of your intestines spill out from a huge cut in your [chest]!"))
+			return
+
+	// Digest the stuff in our stomach, just a bit
+	for (var/atom/movable/thing as anything in stomach_contents)
+		if (SEND_SIGNAL(thing, COMSIG_ATOM_STOMACH_DIGESTED, src, owner, seconds_per_tick) & COMPONENT_CANCEL_DIGESTION)
+			continue
+		var/acid_pwr = stomach_acid_power(thing, seconds_per_tick)
+		if (acid_pwr)
+			thing.acid_act(acid_pwr, 10)
+
+		// If you have strong stomach you can eat glass, literally
+		if (!isitem(thing) || HAS_TRAIT(owner, TRAIT_STRONG_STOMACH))
+			continue
+
+		var/obj/item/as_item = thing
+		// If your stomach is cut open, it will hurt like hell
+		if (cut_open_damage)
+			if (chest && !chest.cavity_item && as_item.w_class <= WEIGHT_CLASS_NORMAL)
+				// Oopsie!
+				chest.cavity_item = as_item
+				stomach_contents -= as_item
+				continue
+
+			owner.apply_damage(as_item.w_class * (as_item.sharpness ? 2 : 1), BRUTE, BODY_ZONE_CHEST, wound_bonus = CANT_WOUND,
+				sharpness = as_item.sharpness, attacking_item = as_item, wound_clothing = FALSE)
+
+		if (!as_item.sharpness)
+			continue
+
+		var/was_failing = (organ_flags & ORGAN_FAILING)
+		apply_organ_damage(as_item.w_class)
+		// Damage caused organ failure
+		if (was_failing || !(organ_flags & ORGAN_FAILING))
+			continue
+
+		cut_open_damage = maxHealth * 0.5
+		if (HAS_TRAIT(owner, TRAIT_ANALGESIA))
+			continue
+
+		owner.visible_message(span_warning("[owner] doubles over in pain!"), span_userdanger("You feel a sharp, searing sensation in your stomach!"))
+		owner.Paralyze(1 SECONDS)
+		owner.adjust_eye_blur(5 SECONDS)
+
+/// Acid power of whatever we're digesting
+/obj/item/organ/stomach/proc/stomach_acid_power(atom/movable/nomnom, seconds_per_tick)
+	if (isliving(nomnom)) // NO VORE ALLOWED
```

Comment:
1984

Reply:
are you sure you want people being able to ***digest*** others?

Reply:
absolutely 100% we need vore